File: free_tasks/new_palindrome.py
class Solution:
    val = 12
    def new_pal(val):
        string_val = str(val)
        reversed_string_val = string_val[::-1]

        final_val = val
        reversed_final_val = 0
        while final_val >= val and final_val != reversed_final_val:    
            string_final_val = str(final_val)
            reversed_string_final_val = string_final_val[::-1]
            reversed_final_val = int(reversed_string_final_val)
            if final_val > val and final_val == reversed_final_val:    
                return final_val
            else:
                final_val += 1
    print(new_pal(val))

File: free_tasks/ft_js/new_palindrome.js
class Solution {
    static newPal(val) {
        const stringVal = val.toString();
        const reversedStringVal = stringVal.split('').reverse().join('');

        let finalVal = val;
        let reversedFinalVal = 0;

        while (finalVal >= val && finalVal !== reversedFinalVal) {
            const stringFinalVal = finalVal.toString();
            const reversedStringFinalVal = stringFinalVal.split('').reverse().join('');
            reversedFinalVal = parseInt(reversedStringFinalVal);

            if (finalVal > val && finalVal === reversedFinalVal) {
                return finalVal;
            } else {
                finalVal++;
            }
        }
    }
}

const val = 12;
console.log(Solution.newPal(val));
